### PR TITLE
add tcp keep alive hack to net5 target

### PR DESCRIPTION
**Pull request by tg123**
this is the temp fix for #533 and net5 only
net5 users will have tcp keepalive by default

### CODE REVIEW

**Comment by tg123** [comment]
#555

File: src/KubernetesClient/Kubernetes.ConfigInit.cs [hunk 1]
```diff
@@ -166,6 +168,27 @@ private void AppendDelegatingHandler<T>()
...
+                    NoDelay = true,
+                };
+
+                socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
+
+                await socket.ConnectAsync(context.DnsEndPoint.Host, context.DnsEndPoint.Port, token).ConfigureAwait(false);
+                return new NetworkStream(socket, ownsSocket: true);
+            };
+
+            var p = HttpClientHandler.GetType().GetField("_underlyingHandler", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (brendandburns) [inline src/KubernetesClient/Kubernetes.ConfigInit.cs:188]: Add a comment here to explain where this magic field name came from

**Comment by brendandburns** [comment]
Looks good, one small comment on documenting things.

(separately, looks like we should exclude tests from the codeql scanning)

**Comment by brendandburns** [comment]
We should perhaps only add keep-alives for sockets used for watch.

**Comment by brendandburns** [comment]
I think we're hitting this:

https://github.com/dotnet/runtime/issues/24917

**Comment by brendandburns** [comment]
Perhaps we should switch to HTTP/2 pings?

https://github.com/dotnet/runtime/pull/40257

**Comment by tg123** [comment]
I tested this when I first impl the tcp keepalive
at first I think it is `tcp keepalive` but I am still being kicked out after enable this.

> Perhaps we should switch to HTTP/2 pings?
> 
> [dotnet/runtime#40257](https://github.com/dotnet/runtime/pull/40257)

**Pull request merged**